file_changed_command_runner/callback_server.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import os
+import threading
+
+from http.server import HTTPServer, BaseHTTPRequestHandler
+
+
+log = logging.getLogger(__name__)
+
+
+class CallbackHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(200, "OK")
+        self.end_headers()
+        self.close_connection = True  # pylint: disable=attribute-defined-outside-init
+
+        self.server: "CallbackServer"
+        self.server.process_callback(self.path.lstrip("/"))
+
+
+class CallbackServer(HTTPServer):
+    """Listens on localhost for a request with a specific token"""
+
+    port_range = range(50000, 51000)
+
+    def __init__(self):
+        self.allow_reuse_address: bool = False
+        self._token: str = ""
+        self._event: threading.Event = threading.Event()
+
+        for port in self.port_range:
+            try:
+                super().__init__(("localhost", port), CallbackHandler)
+                log.info("Listening on localhost:%s", self.server_port)
+                break
+            except OSError:
+                # occurs when a given port is unavailable
+                pass
+
+        if not hasattr(self, "server_port"):
+            raise RuntimeError("Failed to bind to a localhost port")
+
+        threading.Thread(target=self.serve_forever, daemon=True).start()
```

Comment:
It'd be nice to be able to explicitly shut down the server, since otherwise this will keep going until process exit

Reply:
this class is a context manager so makes sense to serve only while in a "waiting" context - made this change